### ISSUE TITLE
openssh-client=9.3_p1-r2 and git=2.40.0-r1 broken in Alpine 3.14 

### ISSUE THREAD

**Issue by isuftin** [alpinelinux/docker-alpine#317]
```sh
docker run --rm -it alpine:3.14 /bin/ash

# apk add --no-cache openssh-client~=9.3 --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
(1/7) Installing libcrypto3 (3.1.0-r3)
(2/7) Installing openssh-keygen (9.3_p1-r2)
(3/7) Installing ncurses-terminfo-base (6.4_p20230401-r4)
(4/7) Installing libncursesw (6.4_p20230401-r4)
(5/7) Installing libedit (20221030.3.1-r1)
(6/7) Installing openssh-client-common (9.3_p1-r2)
(7/7) Installing openssh-client-default (9.3_p1-r2)
Executing busybox-1.33.1-r8.trigger
Executing ca-certificates-20220614-r0.trigger
OK: 16 MiB in 22 packages
/ # ssh
Segmentation fault (core dumped)
```

```sh
docker run --rm -it alpine:3.14 /bin/ash

/ # apk add git~=2.40 --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
(1/10) Installing brotli-libs (1.0.9-r12)
(2/10) Installing libcrypto3 (3.1.0-r3)
(3/10) Installing libunistring (1.1-r1)
(4/10) Installing libidn2 (2.3.4-r1)
(5/10) Installing nghttp2-libs (1.52.0-r1)
(6/10) Installing libssl3 (3.1.0-r3)
(7/10) Installing libcurl (8.0.1-r2)
(8/10) Installing libexpat (2.5.0-r1)
(9/10) Installing pcre2 (10.42-r1)
(10/10) Installing git (2.40.0-r1)
Executing busybox-1.33.1-r8.trigger
Executing ca-certificates-20220614-r0.trigger
OK: 21 MiB in 25 packages
/ # git
Segmentation fault (core dumped)
```

Alpine 3.14 image - 9e179bacf43c

I wonder if libcrypto3 - 3.1.0-r3 which came out a few days ago is the culprit

**Comment by isuftin** [comment]
Still an issue with a newer version of libcrypto...

```sh
docker run --rm -it alpine:3.14 /bin/ash
Unable to find image 'alpine:3.14' locally
3.14: Pulling from library/alpine
f7dab3ab2d6e: Pull complete
Digest: sha256:0f2d5c38dd7a4f4f733e688e3a6733cb5ab1ac6e3cb4603a5dd564e5bfb80eed
Status: Downloaded newer image for alpine:3.14
/ # apk add --no-cache openssh-client~=9.3 --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
140672826891080:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1919:
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/main: Permission denied
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
140672826891080:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1919:
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/community: Permission denied
(1/7) Installing libcrypto3 (3.1.0-r5)
(2/7) Installing openssh-keygen (9.3_p1-r4)
(3/7) Installing ncurses-terminfo-base (6.4_p20230514-r0)
(4/7) Installing libncursesw (6.4_p20230514-r0)
(5/7) Installing libedit (20221030.3.1-r2)
(6/7) Installing openssh-client-common (9.3_p1-r4)
(7/7) Installing openssh-client-default (9.3_p1-r4)
Executing busybox-1.33.1-r8.trigger
OK: 15 MiB in 21 packages
/ # ssh
Segmentation fault (core dumped)
```

**Comment by ncopa** [comment]
those binaries from edge was built against newer versions of musl libc. There is no way to guarantee that binary built with a future version of musl libc (or any library in fact) is able to run with older version of the library.

In general, mixing edge repositories with stable is not supported. I may or may not work. In this case it didn't.

You have the options:

- use whatever version of git and openssh-client that is available for alpine 3.14 (which is EOL now btw...)
- build the specific version of git and openssh you need with alpine 3.14.
- use newer version of alpine

there is nothing we can do to fix this in the alpine docker image.
